app/helpers/application_helper.rb

```diff
@@ -1,27 +1,31 @@
 module ApplicationHelper
-  # rubocop:disable Style/OpenStructUse, Performance/OpenStruct
   USER_COLORS = ["#19063A", "#dce9f3"].freeze
 
-  DELETED_USER = OpenStruct.new(
```

Comment:
Couldn't we have kept this constant around and just changed `name`? Given that we can't get new translations without deploying anyway, I don't think there's anything wrong with this approach unless I'm missing something?

Reply:
Maybe this can work?
``` ruby
  DELETED_USER = OpenStruct.new(
    id: nil,
    darker_color: Color::CompareHex.new(USER_COLORS).brightness,
    username: "[deleted user]",
    name: -> { I18n.t("helpers.application_helper.deleted_user") },
    summary: nil,
    twitter_username: nil,
    github_username: nil,
  )
```

Reply:
Yes, that's exactly what I meant 💯

Reply:
I tried a bit in this direction and I think it must look like this to make other codes believe these constants exist while being i18n-friendly. I have no idea if I have to brace for someone trying to override the `:user` property for this case, though.

``` ruby
  DELETED_USER = Class.new(OpenStruct) do
    def initialize
      super
      self[:id] = nil
      self[:darker_color] = Color::CompareHex.new(USER_COLORS).brightness
      self[:username] = "[deleted user]"
      self[:summary] = nil
      self[:twitter_username] = nil
      self[:github_username] = nil
    end

    def [](key)
      return name if key.to_s == "name"
      super
    end
  
    def name
      I18n.t("helpers.application_helper.deleted_user")
    end
  end.new
```

Which do you think better, to wrap and simulate any of those referred constants or to methodify them and change all their callers? I may be overlooking some easier ways, any input is appreciated.

Reply:
I think it's ok to go for a method, I thought about it the wrong way yesterday as we discussed.

Reply:
This part is currently reworked in https://github.com/forem/forem/pull/16070. I think we should wait for that to get merged, then pull in the changes and override the `name` method to return the translation instead.